src/LibraryViewExtension/Handlers/NodeItemDataProvider.cs

```diff
@@ -76,7 +76,8 @@ protected virtual object CreateObjectForSerialization(IEnumerable<NodeSearchElem
             var data = new LoadedTypeData<LoadedTypeItem>();
             data.loadedTypes = searchEntries
                 //.Where(e => !e.ElementType.HasFlag(ElementTypes.Packaged))
-                .Select(e => CreateLoadedTypeItem<LoadedTypeItem>(e)).ToList();
+                //.Select(e => CreateLoadedTypeItem<LoadedTypeItem>(e)).ToList(); // Excpetion: Collection was modified; enumeration operation may not execute.
+                .ToList().Select(e => CreateLoadedTypeItem<LoadedTypeItem>(e)).ToList();
```

Comment:
@mjkkirschner When I open big files I receive this exception: "Collection was modified; enumeration operation may not execute", so I fixed it by convert that enumeration to list before calling LINQ Select function.

Reply:
this is worrisome - because I don't see where this method would be modifying this collection, which means it's getting modified perhaps by some other thread... are you seeing this in sandbox or revit?

Reply:
neither one nor the other, I'm using Advance Steel. Unfortunately I don't have much time to treat those issues and I'm seeing a lot of bugs on the process of opening Json files, dyn and dyf files. I'm gonna do downgrade dynamo version. But if you need that I do some test in my environment, you may request me.

Reply:
<img width="891" alt="screen shot 2017-10-19 at 19 55 08" src="https://user-images.githubusercontent.com/13578579/31818744-35b56b4c-b578-11e7-8c03-6e30a0aea092.png">

Reply:
@phliberato can you post the stack trace for this exception?

Reply:
not yet, because I did downgrade of dynamo version. Give me some days that I will organize my code to work with both versions.

Reply:
@QilongTang this may cause blank library intermittently.

Reply:
@phliberato can you please paste your sample graphs here that reproduce this issue?

Reply:
@mjkkirschner I'm going to take the stack trace for this exception and then post here, but before that I need to re-update our Dynamo plugin to do my tests. I'm going to try this next week, ok?
I think difficult send to you our graph, because we have a lot of custom nodes into our graphs and many nodes that we have developed especificaly for our demands.

Reply:
yep, understood, you may want to try using the package publish locally function and then send the package over email or another non public method to me.

Reply:
please send me email using my github user + gmail

Reply:
@QilongTang  @jnealb  any idea if this is filed separately?